### DISCUSSION TITLE
limit number of popups

### DISCUSSION THREAD

**Discussion by shalva97** [minbrowser/min#1660] [Ideas]
there are few torrent websites I use to pirate movies and anime. Of course all of them show popups almost every second mouse click. before popups it was peace... now its all these singles in my area, so many :D

**Comment by PalmerAL** [comment]
As I mentioned in Discord, I was hoping we wouldn't need a popup blocker. That clearly didn't work out :)

I suppose the remaining question is which approach to use for this:
* Opt-in - when a site tries to open a popup for the first time, you get a prompt to allow popups for that site
* Opt-out - you can disable popups for certain sites in settings

Not sure which one is a better approach here.

**Reply by shalva97** [reply]
Opt-in looks nice. But I think the problem is not the popups, because before few tabs would open as about#blocked, which was also not that often. so are they actual popups?

the website was torrentgalaxy
![image](https://user-images.githubusercontent.com/22417494/123081885-7c5db700-d42f-11eb-996d-4913f7b31775.png)

that thing annoyed me, they put an invisible link over the page and it doesnot seems to be a popup but a simple link, right? it also triggers by right click.

now im a bit confused what popups exactly are... coming from Firefox my guess is that popups are websites that open in separate window without navigation buttons, but in Min everything opens in new tabs so how to distinguish them?

**Reply by PalmerAL** [reply]
Popups are pages that the site opens by calling `window.open`. The link in the screenshot wouldn't be considered a popup; it's possible the popup is coming from somewhere else though.

Min didn't support popups very well in the past, so if a website tried to open a popup, you would sometimes get a new tab with about:blank instead of the actual popup URL. Now opening the popup actually works, so you get the page the site intended to open instead.

> coming from Firefox my guess is that popups are websites that open in separate window without navigation buttons, but in Min everything opens in new tabs so how to distinguish them?

Yup that's right. It's true you can't tell popups apart from normal tabs in Min; I think in most cases you shouldn't need to though.

**Reply by PalmerAL** [reply]
Also, @lubiedo proposed a pretty nice approach that uses Easylist's adblocking rules to block popups: https://discord.com/channels/764269005195968512/764560513668612166/856953236384383027

If you get a chance to try it, it would be interesting to see whether that works on the site you mentioned.

**Reply by lubiedo** [reply]
Lost my previous branch but tried to replicate it in this PR: #1674
